gitlab_templates/build_ws_image.yml

```diff
@@ -0,0 +1,39 @@
+stages:
+  - build
+
+build-ws:
+  stage: build
+  variables:
+    IMAGE_TAG: "datafed/ws"
+    GIT_STRATEGY: clone
+  tags:
+    - ci-datafed-core
+    - docker
+  rules:
+    - changes:
+        - docker/**/*
+        - scripts/**/*
+        - web/**/*
+        - common/proto/**/*
+        - .gitlab-ci.yml
+      when: on_success
+  script:
+    - BRANCH_LOWER=$(echo "$CI_COMMIT_REF_NAME" | tr '[:upper:]' '[:lower:]')
+    - echo "$BRANCH_LOWER"
+    - ./scripts/generate_datafed.sh
+    - docker login "${REGISTRY}" -u "${HARBOR_USER}" -p "${HARBOR_DATAFED_GITLAB_CI_REGISTRY}"
+    - docker build --build-arg DEPENDENCIES="${REGISTRY}/datafed/dependencies-${BRANCH_LOWER}:latest" --build-arg RUNTIME="${REGISTRY}/datafed/runtime-${BRANCH_LOWER}:latest" -f web/docker/Dockerfile -t "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}:latest" .
+    - docker push "${REGISTRY}/${IMAGE_TAG}-${BRANCH_LOWER}"
+
+# Prevents downstream pipeline from being empty and triggering a failure, the
+# build-ws job above may or may not run because of the rules section, if there
+# are no jobs in the pipeline GitLab has the unfortunate behavior of reporting
+# the trigger as a failure. To prevent the pipeline from being empty the
+# placeholder job is created.
```

Comment:
Looks like this is the problem you're describing 😞 : https://gitlab.com/gitlab-org/gitlab/-/issues/419293
I see a proposal [here](https://gitlab.com/groups/gitlab-org/-/epics/11447#proposal). I'd like you to see if any action was taken on this so we can avoid the placeholder and keep this nice work clean

Reply:
Good catch.

Reply:
Looks like strategy: mirror is not yet supported in our version of GitLab. Maybe we can refactor at some point in the future.

They don't even have it in their official documentation yet. 

https://docs.gitlab.com/ee/ci/pipelines/downstream_pipelines.html

![Screenshot (16)](https://github.com/user-attachments/assets/38a286ff-ba55-4fd3-bcf9-effa61e65ad5)